src/OptionSettlement.sol

```diff
@@ -208,7 +206,33 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
             revert EncodedOptionIdInClaimIdDoesNotMatchProvidedOptionId(claimId, optionId);
         }
 
-        Option storage optionRecord = _writeOptions(_optionIdU160b, amount);
+        if (amount == 0) {
+            revert AmountWrittenCannotBeZero();
+        }
+
+        Option storage optionRecord = _option[_optionIdU160b];
```

Comment:
there will be some conflicts here, so just keep it in mind that we'd need to do another pass after #69 to inline where needed